tests/test.bats

```diff
@@ -0,0 +1,340 @@
+  local -r username="$2"
+  local -r password="$3"
+
+  echo "${cmd}" | docker exec -i oracle sqlplus -S "${username}/${password}@ORCLPDB1"
+}
+
+# docker_sqlplus_admin executes an Oracle sqlplus query as sysdba. This should
+# only be used for DB admin tasks such as creating a new static user.
+#
+# Parameters:
+#   cmd
+docker_sqlplus_admin() {
+  local -r cmd="$1"
+  echo "${cmd}" | docker exec -i oracle sqlplus -S "sys/$VPASS@ORCLPDB1" as sysdba
+}
+
+# setup_file runs once before all tests
+setup_file(){
+  # clear log file
+  echo "" > $SETUP_TEARDOWN_OUTFILE
+  echo "" > $VAULT_OUTFILE
+
+  VAULT_TOKEN='root'
+
+  log "BEGIN SETUP"
+
+  docker rm oracle --force
+  docker run \
+    -d \
+    --rm \
+    --name oracle \
+    -p 1521:1521 \
+    -p 5500:5500 \
+    -e ORACLE_SID=myservice \
+    -e ORACLE_PWD=$VPASS \
+    -v $DOCKER_VOLUME_MNT:/opt/oracle/oradata \
+    oracle/database:19.3.0-se2
+
+  log "waiting for oracle db..."
+  while ! docker logs oracle | grep "DATABASE IS READY TO USE" > /dev/null; do sleep 1; done
+
+  # setup vault user
+  docker_sqlplus_admin "DROP USER $VAULT_USER;"
+  docker_sqlplus_admin "CREATE USER $VAULT_USER IDENTIFIED BY \"$VPASS\";"
+  docker_sqlplus_admin "GRANT ALL PRIVILEGES TO $VAULT_USER;"
+  docker_sqlplus_admin "GRANT SELECT ON gv_\$SESSION TO $VAULT_USER;"
+
+  ./vault server -dev -dev-root-token-id=root -log-level=trace \
+    -dev-plugin-dir=${PLUGIN_DIR?} > $VAULT_OUTFILE 2>&1 &
+
+  log "waiting for vault..."
+  i=0
+  while ! vault status >/dev/null 2>&1; do
+    sleep 1
+    ((i=i+1))
+    [ $i -gt $VAULT_STARTUP_TIMEOUT ] && log_err "timed out waiting for vault to start"
+  done
+
+  vault login ${VAULT_TOKEN?}
+
+  run vault status
+  assert_status 0
+  log "vault started successfully"
+
+  vault namespace create ns1
+
+  vault secrets enable --namespace=ns1 database
+  log "vault enabled database secrets engine"
+
+  log "END SETUP"
+}
+
+# teardown_file runs once after all tests complete
+teardown_file(){
+  log "BEGIN TEARDOWN"
+
+  log "dropping the vault user from oracle db"
+  docker_sqlplus_admin "DROP USER $VAULT_USER;"
+
+  log "dropping the static user from oracle db"
+  docker_sqlplus_admin "DROP USER $STATIC_USER;"
+
+  log "removing the oracle docker container"
+  docker rm oracle --force
+
+  log "killing vault process"
+  pkill vault
+
+  log "END TEARDOWN"
+}
+
+@test "Read license" {
+  run vault read -format=json sys/license/status
+  assert_status 0
+}
+
+@test "POST /database/config/:name - write oracle connection config" {
+  run vault write --namespace=ns1 database/config/$DB_NAME \
+    plugin_name=vault-plugin-database-oracle \
+    connection_url="{{username}}/{{password}}@localhost:1521/ORCLPDB1" \
+    allowed_roles="*" \
+    username="$VAULT_USER" \
+    password="$VPASS" \
+    max_conneciton_lifetime="30s"
+  assert_status 0
+}
+
+@test "GET /database/config/:name - read oracle connection config" {
+  run vault read --namespace=ns1 database/config/$DB_NAME
```

Comment:
Might be nice to have some assertions that the contents are correct too? You could use `-format=json` and do some processing with `jq`.

Reply:
will do thanks!